config/application.rb

```diff
@@ -107,113 +102,63 @@ def config.database_configuration
     # issue is image_optim crashes on missing dependencies
     config.assets.image_optim = false
 
-    config.autoloader = :zeitwerk
+    require 'discourse_inflector'
+    Rails.autoloaders.each do |autoloader|
+      autoloader.inflector = DiscourseInflector.new
+      autoloader.inflector.inflect(
+        'canonical_url' => 'CanonicalURL',
+        'clean_up_unmatched_ips' => 'CleanUpUnmatchedIPs',
+        'homepage_constraint' => 'HomePageConstraint',
+        'ip_addr' => 'IPAddr',
+        'onpdiff' => 'ONPDiff',
+        'onceoff' => 'Jobs',
+        'pop3_polling_enabled_setting_validator' => 'POP3PollingEnabledSettingValidator',
+        'regular' => 'Jobs',
+        'scheduled' => 'Jobs',
+        'version' => 'Discourse',
+        'csrf_token_verifier' => 'CSRFTokenVerifier',
+        'html' => 'HTML',
+        'google_oauth2_authenticator' => 'GoogleOAuth2Authenticator',
+        'oauth2_authenticator' => 'OAuth2Authenticator',
+        'omniauth_strategies' => 'OmniAuthStrategies',
+        'json' => 'JSON',
+      )
+    end
 
     # Custom directories with classes and modules you want to be autoloadable.
-    config.autoload_paths += Dir["#{config.root}/app"]
-    config.autoload_paths += Dir["#{config.root}/app/jobs"]
-    config.autoload_paths += Dir["#{config.root}/app/serializers"]
-    config.autoload_paths += Dir["#{config.root}/lib"]
-    config.autoload_paths += Dir["#{config.root}/lib/common_passwords"]
-    config.autoload_paths += Dir["#{config.root}/lib/highlight_js"]
-    config.autoload_paths += Dir["#{config.root}/lib/i18n"]
-    config.autoload_paths += Dir["#{config.root}/lib/validators/"]
-
-    Rails.autoloaders.main.ignore(Dir["#{config.root}/app/models/reports"])
-    Rails.autoloaders.main.ignore(Dir["#{config.root}/lib/freedom_patches"])
-
-    def watchable_args
-      files, dirs = super
-
-      # Skip the assets directory. It doesn't contain any .rb files, so watching it
-      # is just slowing things down and raising warnings about node_modules symlinks
-      app_file_extensions = dirs.delete("#{config.root}/app")
-      Dir["#{config.root}/app/*"].reject { |path| path.end_with? "/assets" }.each do |path|
-        dirs[path] = app_file_extensions
-      end
-
-      [files, dirs]
-    end
+    config.autoload_paths << "#{root}/lib"
+    config.autoload_paths << "#{root}/lib/common_passwords"
+    config.autoload_paths << "#{root}/lib/highlight_js"
+    config.autoload_paths << "#{root}/lib/i18n"
+    config.autoload_paths << "#{root}/lib/validators"
+    config.autoload_paths << "#{root}/lib/svg_sprite"
+    config.autoload_paths << "#{root}/lib/guardian"
+
+    Rails.autoloaders.main.ignore("app/models/reports",
```

Comment:
Why are there more ignores?

Reply:
Basically because the autoloader now works as expected :sweat_smile: The files listed here are not following the Rails conventions (where the module/class name should match its path) so we need to tell Zeitwerk to ignore those. What we could do as well is to move/rename these modules/classes to follow the Rails conventions.

Reply:
> What we could do as well is to move/rename these modules/classes to follow the Rails conventions.

How hard would that be?

Reply:
I don’t think that’s very hard, more like time consuming and it will add more stuff to review to this PR. But yes I agree with you :+1:

Reply:
If there are no other drawbacks other than time, then can you please do it?

Reply:
Sure, will do then!